internal/pkg/network/dhcp.go

```diff
@@ -118,6 +120,13 @@ func (service *Service) dhclient4(ctx context.Context, ifname string, modifiers
 				hostname = *kernHostname
 			}
 
+			// Truncate hostname to be betta
+			// Allow IP addrs to be valid hostnames for the time being
+			if ok := net.ParseIP(hostname); ok != nil {
+				// Pull out the first part of a potential FQDN
+				hostname = strings.Split(hostname, ".")[0]
```

Comment:
to be on the safe side, I would check first that hostname is not empty, as split returns empty slice in that case

https://play.golang.org/p/3BS4TfAMXxT

Reply:
sorry, I confused myself, it's still one-item slice.